allennlp/training/trainer.py

```diff
@@ -788,57 +811,26 @@ def from_params(  # type: ignore
             model = model.cuda(cuda_device)
 
         parameters = [[n, p] for n, p in model.named_parameters() if p.requires_grad]
-        optimizer = Optimizer.from_params(parameters, params.pop("optimizer"))
-        if "moving_average" in params:
-            moving_average = MovingAverage.from_params(
-                params.pop("moving_average"), parameters=parameters
-            )
+        optimizer = optimizer.construct(parameters=parameters)
+        if moving_average:
+            moving_average_ = moving_average.construct(parametrs=parametrs)
         else:
-            moving_average = None
+            moving_average_ = None
 
-        if lr_scheduler_params:
-            lr_scheduler = LearningRateScheduler.from_params(optimizer, lr_scheduler_params)
-        else:
-            lr_scheduler = None
-        if momentum_scheduler_params:
-            momentum_scheduler = MomentumScheduler.from_params(optimizer, momentum_scheduler_params)
+        if learning_rate_scheduler:
+            learning_rate_scheduler_ = learning_rate_scheduler.construct(optimizer)
         else:
-            momentum_scheduler = None
+            learning_rate_scheduler_ = None
 
-        if "checkpointer" in params:
-            if (
-                "keep_serialized_model_every_num_seconds" in params
-                or "num_serialized_models_to_keep" in params
-            ):
-                raise ConfigurationError(
-                    "Checkpointer may be initialized either from the 'checkpointer' key or from the "
```

Comment:
Note that I removed this duplicate means of specifying the parameters - you now have to pass the arguments to the checkpointer.